### PR TITLE
Matplotlib versions

### PR DESCRIPTION
**Pull request by lldelisle**
Be more flexible on matplotlib versions.

### CODE REVIEW

**Comment by alexlenail** [comment]
I'm surprised the tests pass / you aren't seeing the same bug I was seeing in #356

**Comment by lldelisle** [comment]
No, I don't understand how you got this error...
I think the only place we use grid is:
https://github.com/deeptools/pyGenomeTracks/blob/488a6d3f9e81c6ca1af9f90484445070ed9c7024/pygenometracks/utilities.py#L201-L202

**Comment by lldelisle** [comment]
Could you tell me how you got this error? PGT version ? matplotlib version ? Ini file ?

**Comment by lldelisle** [comment]
> I'm surprised the tests pass / you aren't seeing the same bug I was seeing in #356

Our tests are not exhaustive...

**Comment by alexlenail** [comment]
pyGenomeTracks==3.6
matplotlib 3.4.3

```ini
[x-axis]

[spacer]

[GRCh38_V29]
file = data/raw/genome/GRCh38_V29/GRCh38_V29.gtf
title = GRCh38_V29
file_type = bed
style = UCSC
height = 10
```

File: docs/content/installation.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ Python dependencies:
 * pyBigWig >= 0.3.16
 * hicmatrix >= 15
 * pysam >= 0.14
-* matplotlib >=3.1.1,<=3.3.2
+* matplotlib >=3.1.1,<=3.4.3
```
Comment (joachimwolff) [inline docs/content/installation.rst:21]: You define in the test matrix a matplotlib version 3.5.1, but restrict it here to 3.4.3?

File: environment.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ channels:
    - conda-forge
 dependencies:
     - numpy >=1.16
-    - matplotlib >=3.1.1,<=3.2.2
+    - matplotlib >=3.1.1,<=3.4.3
```
Comment (joachimwolff) [inline environment.yml:10]: Again matplotlib version

File: setup.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def checkProgramIsInstalled(self, program, args, where_to_download,
 
 
 install_requires_py = ["numpy >=1.16",
-                       "matplotlib >=3.1.1,<=3.3.2",
+                       "matplotlib >=3.1.1,<=3.4.3",
```
Comment (joachimwolff) [inline setup.py:94]: Again matplotlib version

**Comment by lldelisle** [comment]
Good catch.
I modify it.

File: docs/content/installation.rst (hunk 1 above)
Comment (lldelisle) [reply docs/content/installation.rst]: done, thanks

File: setup.py (hunk 3 above)
Comment (lldelisle) [reply setup.py]: done, thanks

File: environment.yml (hunk 2 above)
Comment (lldelisle) [reply environment.yml]: done, thanks

**Review by joachimwolff — APPROVED** [review]

**Pull request merged**